File: backend/src/main/java/com/team05/petmeeting/domain/user/dto/profile/UserSummaryRes.java
package com.team05.petmeeting.domain.user.dto.profile;

import com.team05.petmeeting.domain.user.entity.User;

public record UserSummaryRes(
        String nickname,
        String name,
        String profileImageUrl
) {
    public static UserSummaryRes from(User user) {
        return new UserSummaryRes(
                user.getNickname(),
                user.getRealname(),
                user.getProfileImageUrl()
        );
    }
}

File: backend/src/main/java/com/team05/petmeeting/domain/user/dto/profile/UserSummaryRes.kt
package com.team05.petmeeting.domain.user.dto.profile

import com.team05.petmeeting.domain.user.entity.User

data class UserSummaryRes(
    val nickname: String,
    val name: String,
    val profileImageUrl: String
) {
    companion object {
        @JvmStatic
        fun from(user: User): UserSummaryRes {
            return UserSummaryRes(
                user.nickname,
                user.realname,
                user.profileImageUrl
            )
        }
    }
}
